### PR TITLE
Created a unit test for the class ItemDataExtractor.

### PR DESCRIPTION
**Pull request by JEntressangle**

### CODE REVIEW

File: tests/auto/test_itemdataextractor.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <QtTest/QtTest>
```
Comment (dfaure-kdab) [inline tests/auto/test_itemdataextractor.cpp]: Ah, I hadn't noticed this before: this include all of QtCore too, oops. Please include <QTest> only, and add any missing includes.

File: tests/auto/test_itemdataextractor.cpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    Akonadi::Item item;
+    item.setPayload<KABC::Addressee>(contact);
+    return item;
+}
+
+class TestItemDataExtractor : public QObject
+{
+    Q_OBJECT
+private Q_SLOTS:
+
```
Comment (dfaure-kdab) [inline tests/auto/test_itemdataextractor.cpp]: move empty line above private Q_SLOTS

File: tests/auto/test_itemdataextractor.cpp [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    void shouldReturnCorrectUrlFromId()
+    {
+        //GIVEN
+        QFETCH(QString, resourceBaseUrl);
+        QFETCH(QString, itemId);
+        QFETCH(DetailsType, type);
+        QFETCH(QUrl, expectedUrl);
+        ItemDataExtractor *itemDE = ItemDataExtractor::createDataExtractor(type, 0);
+        //WHEN
+        QUrl url = itemDE->itemUrl(resourceBaseUrl, itemId);
```
Comment (dfaure-kdab) [inline tests/auto/test_itemdataextractor.cpp]: const QUrl url =

File: tests/auto/test_itemdataextractor.cpp [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    void shouldReturnCorrectUrlFromAkonadiItem()
+    {
+        //GIVEN
+        ItemDataExtractor *itemDE = ItemDataExtractor::createDataExtractor(DetailsType::Account, 0);
+        SugarAccount account;
+        account.setId("1");
+        Akonadi::Item item;
+        item.setPayload<SugarAccount>(account);
+        //WHEN
+        QUrl url = itemDE->itemUrl("KDAB", item);
```
Comment (dfaure-kdab) [inline tests/auto/test_itemdataextractor.cpp]: const

File: tests/auto/test_itemdataextractor.cpp [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    void shouldReturnCorrectId()
+    {
+        //GIVEN
+        QFETCH(DetailsType, type);
+        QFETCH(Akonadi::Item, item);
+        ItemDataExtractor *itemDE = ItemDataExtractor::createDataExtractor(type, 0);
+        //WHEN
+        QString id = itemDE->idForItem(item);
```
Comment (dfaure-kdab) [inline tests/auto/test_itemdataextractor.cpp]: const

**Comment by JEntressangle** [comment]
I made the requested changes.

File: CMakeLists.txt [hunk 6]
```diff
@@ -3,6 +3,7 @@ project(FatCRM)
 cmake_minimum_required(VERSION 2.8.12) # Feature: PUBLIC keyword in target_link_libraries
 if(COMMAND cmake_policy)
   cmake_policy(SET CMP0003 NEW)
+  cmake_policy(SET CMP0059 OLD)
```
Comment (dfaure-kdab) [inline CMakeLists.txt]: oops, I forgot this bit:

if(POLICY CMP0059)
   cmake_policy(SET CMP0059 OLD) # we use ``DEFINITIONS`` as a built-in directory property
endif()

**Pull request merged**